### PR TITLE
feat(proto): add parser

### PR DESCRIPTION
**Pull request by fsouza**
Related to #2307.

First time doing this, so let me know if I did anything wrong. I did notice that the parser has some [queries predefined](https://github.com/mitchellh/tree-sitter-proto/tree/main/queries), so I wonder if we could use those?

### CODE REVIEW

**Review by theHamsta — APPROVED** [review]
Nice! Finally a proto parser that seems to work! Thanks also for creating the highlights.

Just a few comments

File: queries/proto/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "returns"
+  "message"
+  "enum"
+  "oneof"
+  "repeated"
+  "reserved"
+  "to"
+] @keyword
+
+[
```
Comment (theHamsta) [inline queries/proto/highlights.scm:17]: `"\"proto3\""` could be `@string`, `=` and `;` could be highlighted.

Enum identifiers could be highlighted as `@constant` but not a must

File: queries/proto/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  (enum_name)
+  (service_name)
+  (rpc_name)
+] @type
+
+(string) @string
+
+[
+  (int_lit)
+  (float_lit)
```
Comment (theHamsta) [inline queries/proto/highlights.scm:30]: there is a separate highlight for `@float`

File: queries/proto/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+] @number
+
+[
+  (true)
+  (false)
+] @constant.builtin
```
Comment (theHamsta) [inline queries/proto/highlights.scm:36]: `@boolean`?

File: queries/proto/highlights.scm (hunk 1 above)
Comment (fsouza) [reply queries/proto/highlights.scm:17]: Pushed all suggestions except for `=`. Do you think we should use operator for that?

Comment (theHamsta) [reply queries/proto/highlights.scm:17]: I think `@operator` should be good for that

File: queries/proto/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+(comment) @comment
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
```
Comment (theHamsta) [inline queries/proto/highlights.scm:51]: `"<"  `">"` could belong to that category

File: queries/proto/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+]  @punctuation.bracket
+
+
+";" @punctuation.delimiter
```
Comment (theHamsta) [inline queries/proto/highlights.scm:55]: also `","` (use in `map<foo,bar>`

Comment (theHamsta) [reply queries/proto/highlights.scm:55]: I'm unsure about `map` itself. Leave it like it is or use `@type`?

Comment (theHamsta) [reply queries/proto/highlights.scm:55]: `(message_or_enum_type) @type`

**Comment by fsouza** [comment]
@theHamsta I think I got it all, let me know if I missed anything or if there's any new feedback!

**Comment by theHamsta** [comment]
@fsouza No, it's great. Thank you a lot for this contribution. I've been waiting a long time for that. Previous proto-parsers I found couldn't reliably parse most files.

**Review by theHamsta — APPROVED** [review]

**Pull request merged**